lib/src/widget/asset_picker.dart

```diff
@@ -134,6 +140,25 @@ class AssetPickerState<Asset, Path> extends State<AssetPicker<Asset, Path>>
     });
   }
 
+  Future<void> _checkAndRequestPermission() async {
+    if (!_shouldRequestPermission) {
+      return;
+    }
+
+    _shouldRequestPermission = false;
+
+    final completer = Completer<void>();
```

Comment:
Completer inside the method doesn't make sense because the control flow still relies on the `_shouldRequestPermission` flag.

Reply:
Completer has removed out from the method. Please advice in details if there is any additional changes required.

Reply:
You don't need `_shouldRequestPermission` anymore, no?

Reply:
`_shouldRequestPermission` is still required because `didChangeAppLifecycleState` is called after completer completed. For example, after `requestPermissionExtend` return result, completer will then be completed, eventually `didChangeAppLifecycleState` with `AppLifecycleState.resumed` state will be invoked, causing loop.

Reply:
Every time the app gets resumed, the permission status can be changed. Using the flag will potentially lead to inconsistent results of permissions.

Reply:
I have removed the `_shouldRequestPermission` flag, the logic now fully depends on completer.